SpatialGDK/Documentation/content/tutorials/multiserver-shooter/tutorial-multiserver-healthchanges.md

```diff
@@ -1,83 +1,82 @@
 
-1. In your IDE, open `UnrealGDKThirdPersonShooter\Game\Source\ThirdPersonShooter\Characters\TPSCharacter.cpp`.
+1. In your IDE, open `UnrealGDKExampleProject\Game\Source\GDKShooter\Private\Characters\Components\HealthComponent.cpp`.
 1. Navigate to the `GetLifetimeReplicatedProps` function (which is implementd around line 182), and insert the following snippet:
 
 ```
     // Only replicate health to the owning client.
-    DOREPLIFETIME_CONDITION(ATPSCharacter, CurrentHealth, COND_OwnerOnly);
+    DOREPLIFETIME(UHealthComponent, CurrentHealth);
 ```
 
-    > **Note:** You only want to replicate the `CurrentHealth` variable to the client that owns this Actor, thus you specify the `COND_OwnerOnly` flag.
-    
     Finally, you need to implement the `OnRep_CurrentHealth` function so that the player health UI gets updated when the `CurrentHealth` variable is replicated:
 
-1. In your IDE, open `UnrealGDKThirdPersonShooter\Game\Source\ThirdPersonShooter\Characters\TPSCharacter.h`.
+1. In your IDE, open `UnrealGDKExampleProject\Game\Source\GDKShooter\Public\Characters\Components\HealthComponent.h`.
 1. In the public scope of the class, insert the following snippet:
 
     ```	
         UFUNCTION()
         void OnRep_CurrentHealth();
     ```
 
-1. In your IDE, open `UnrealGDKThirdPersonShooter\Game\Source\ThirdPersonShooter\Characters\TPSCharacter.cpp` and insert the following snippet:
+1. In your IDE, open `UnrealGDKExampleProject\Game\Source\GDKShooter\Private\Characters\Components\HealthComponent.cpp` and insert the following snippet:
 
 ```
-    void ATPSCharacter::OnRep_CurrentHealth()
+    void UHealthComponent::OnRep_CurrentHealth()
     {
-    	if (GetNetMode() != NM_DedicatedServer)
-    	{
-    	ATPSPlayerController* PC = Cast<ATPSPlayerController>(GetController());
-    		if (PC)
-    		{
-    			PC->UpdateHealthUI(CurrentHealth, MaxHealth);
-    		}
-    		else
-    		{
-    			UE_LOG(LogTPS, Warning, TEXT("Couldn't find a player controller for character: %s"), *this->GetName());
-    		}
-    	}
+	    HealthUpdated.Broadcast(CurrentHealth, MaxHealth);
+	    if (CurrentHealth <= 0.f)
+	    {
+		    Death.Broadcast();
+	    }
     }
 ```
 
 </br>
 Notice that the workflow you just used mirrors that of native Unreal.
 
-### Step 2: Rebuild your project
-Because you have changed code in a function, you now need to rebuild your project. Additionally, because you've modified code related to replication, you need to generate schema. To do this:
+### Step 2: Generate schema and rebuild your project
+Because you've modified code related to replication, SpatialOS requires you to generate schema. Also, because you have changed code in a GDK function, Unreal Engine requires you to rebuild your project. (Note: SpatialOS does not require you to generate a snapshot, this is because as you did this when you set up the Example Project and you only need to set up a snapshot once per project.) </br>
```

Comment:
```suggestion
Because you've modified code related to replication, SpatialOS requires you to generate schema. Also, because you have changed code in your project, Unreal Engine requires you to rebuild your project. (Note: SpatialOS does not require you to generate a snapshot, this is because as you did this when you set up the Example Project and you only need to set up a snapshot once per project.) </br>
```

Reply:
added it in.